ratbagd/ratbagd-profile.c

```diff
@@ -347,13 +347,103 @@ ratbagd_profile_is_enabled(sd_bus *bus,
 	return sd_bus_message_append(reply, "b", enabled);
 }
 
+static int
+ratbagd_profile_set_name(sd_bus *bus,
+			 const char *path,
+			 const char *interface,
+			 const char *property,
+			 sd_bus_message *m,
+			 void *userdata,
+			 sd_bus_error *error)
+{
+	struct ratbagd_profile *profile = userdata;
+	char *name;
+	int r;
+
+	r = sd_bus_message_read(m, "s", &name);
+	if (r < 0)
+		return r;
+
+	r = ratbag_profile_set_name(profile->lib_profile, name);
+
+	if (r == 0) {
+		sd_bus_emit_properties_changed(bus,
+					       profile->path,
+					       RATBAGD_NAME_ROOT ".Profile",
+					       "Name",
+					       NULL);
+	}
+
+	/*
+	 * Note: we should return `r`, but the d-bus bindings doesn't
+	 * like non standard errors. So just return success and let the client
+	 * check for the capability before-hand and let .commit() signal an
+	 * error.
+	 */
+	return 0;
```

Comment:
This is different from how the rest of ratbagd works at the moment; are you sure this is correct? AFAIK these errors are received and processed properly by the bindings. Otherwise, we should update all of ratbagd.

Reply:
Well, the error is processed, forwarded to the binding and then ignored by it (a stacktrace is shown, but that's it). Given that the gobject setters can't forward exceptions, we'd better shut up the warning (or there is duplicated messages that will confuse users).

Agree, we should change all the binding to reply either success or a non-libratbag error.

We should first try to re-merge piper and libratbag first.

Reply:
> Well, the error is processed, forwarded to the binding and then ignored by it (a stacktrace is shown, but that's it).

Indeed, exceptions should be caught be the user of the bindings. I see what you mean now with `GDBus.Error:org.freedesktop.DBus.Error.Failed: Unknown error 1001 (0)`; indeed an optimistic approach that checks for errors on `Commit()` seems the best way out.

> Given that the gobject setters can't forward exceptions, we'd better shut up the warning (or there is duplicated messages that will confuse users).

This is resolved in the latest version of the bindings, where no GObject setter uses `_dbus_call` anymore.